cdap-formats/src/test/java/io/cdap/cdap/format/StructuredRecordStringConverterTest.java

```diff
@@ -160,6 +167,57 @@ public void testTimestampLogicalTypeConversion() throws Exception {
     assertRecordsEqual(record, recordOfJson);
   }
 
+  @Test
+  public void testDatetimeLogicalTypeConversion() throws Exception {
+    Schema dateSchema = Schema.recordOf(
+      "date",
+      Schema.Field.of("id", Schema.of(Schema.Type.INT)),
+      Schema.Field.of("name", Schema.nullableOf(Schema.of(Schema.Type.STRING))),
+      Schema.Field.of("datetimefield", Schema.nullableOf(Schema.of(Schema.LogicalType.DATETIME))));
+
+    StructuredRecord record = StructuredRecord.builder(dateSchema).set("id", 1)
+      .set("name", "alice")
+      .setDateTime("datetimefield", LocalDateTime.now()).build();
+
+    String jsonOfRecord = StructuredRecordStringConverter.toJsonString(record);
+    StructuredRecord recordOfJson = StructuredRecordStringConverter.fromJsonString(jsonOfRecord, dateSchema);
+
+    assertRecordsEqual(record, recordOfJson);
+  }
+
+  @Test
+  public void testInvalidDatetimeLogicalTypeConversion() throws Exception {
+    Schema dateSchema = Schema.recordOf(
+      "date",
+      Schema.Field.of("id", Schema.of(Schema.Type.INT)),
+      Schema.Field.of("name", Schema.nullableOf(Schema.of(Schema.Type.STRING))),
+      Schema.Field.of("datetime_field", Schema.nullableOf(Schema.of(Schema.LogicalType.DATETIME))));
+
+    StructuredRecord record = StructuredRecord.builder(dateSchema).set("id", 1)
+      .set("name", "alice")
+      .set("datetime_field", "random").build();
+    thrown.expect(IOException.class);
+    thrown.expectMessage("Unsupported format. DateTime type should be in ISO-8601 format.");
+    StructuredRecordStringConverter.toJsonString(record);
+  }
+
+  @Test
+  public void testInvalidDatetimeLogicalTypeJSON() throws Exception {
+    Schema dateSchema = Schema.recordOf(
+      "date",
+      Schema.Field.of("id", Schema.of(Schema.Type.INT)),
+      Schema.Field.of("name", Schema.nullableOf(Schema.of(Schema.Type.STRING))),
+      Schema.Field.of("datetime_field", Schema.nullableOf(Schema.of(Schema.LogicalType.DATETIME))));
+    String json = "{\"id\":1,\"name\":\"alice\",\"datetime_field\":\"2021-01-27T02:25:52.088\"}";
```

Comment:
Not a fan of using string to construct json string. It is better to create a Map and the use Gson.toJson.